brage-import/src/main/java/no/sikt/nva/brage/migration/merger/CristinImportPublicationMerger.java

```diff
@@ -127,6 +137,11 @@ private PublicationInstance<? extends Pages> determincePublicationInstance(Refer
             case ReportWorkingPaper reportWorkingPaper -> ReportWorkingPaperMerger.merge(reportWorkingPaper, newPublicationInstance);
             case ReportBookOfAbstract reportBookOfAbstract -> ReportBookOfAbstractMerger.merge(reportBookOfAbstract, newPublicationInstance);
             case ReportBasic reportBasic -> ReportBasicMerger.merge(reportBasic, newPublicationInstance);
+            case JournalIssue journalIssue -> JournalIssueMerger.merge(journalIssue, newPublicationInstance);
```

Comment:
if merge is part of base class, you can call merge without switch here maybe?

Reply:
Have refactored it using inheritance, but it seems like we still need factory method to initiate right publicationInstanceMerger<?>

`return PublicationInstanceMerger.of(publicationInstance).merge(newPublicationInstance);`

```
    public static PublicationInstanceMerger<?> of(PublicationInstance<?> instance) {
        return switch (instance) {
            case DegreePhd degreePhd -> new DegreePhdMerger(degreePhd);
            case DegreeBachelor degreeBachelor -> new DegreeBachelorMerger(degreeBachelor);
            case DegreeMaster degreeMaster -> new DegreeMasterMerger(degreeMaster);
            case DegreeLicentiate degreeLicentiate -> new DegreeLicentiateMerger(degreeLicentiate);
            case OtherStudentWork otherStudentWork -> new OtherStudentWorkMerger(otherStudentWork);
            case ConferenceReport conferenceReport -> new ConferenceReportMerger(conferenceReport);
            case ReportResearch reportResearch -> new ReportResearchMerger(reportResearch);
            case ReportWorkingPaper reportWorkingPaper -> new ReportWorkingPaperMerger(reportWorkingPaper);
            case ReportBookOfAbstract reportBookOfAbstract -> new ReportBookOfAbstractMerger(reportBookOfAbstract);
            case ReportBasic reportBasic -> new ReportBasicMerger(reportBasic);
            case JournalIssue journalIssue -> new JournalIssueMerger(journalIssue);
            case JournalLeader journalLeader -> new JournalLeaderMerger(journalLeader);
            case ProfessionalArticle professionalArticle -> new ProfessionalArticleMerger(professionalArticle);
            case AcademicArticle academicArticle -> new AcademicArticleMerger(academicArticle);
            case MediaFeatureArticle mediaFeatureArticle -> new MediaFeatureArticleMerger(mediaFeatureArticle);
            default -> new NoMerger<>(instance);
        };
    }
```

But would be interesting to see how it can be done without factory method :) If you have some ideas

Reply:
also had a look at it, but sorry to say I had to throw in the towel here. I can't think if an easy and dynamic way to avoid this without using reflection and reflection is probably not worth it in this case.